src/CMakeLists.txt

```diff
@@ -32,23 +41,34 @@ execute_process(
 
 message(STATUS "Awkward includes : ${AWKWARD_INCLUDE}")
 message(STATUS "Awkward libraries: ${AWKWARD_LIBRARIES}")
-include_directories(BEFORE "${AWKWARD_INCLUDE}")
 
 find_library(CPU-KERNELS awkward-cpu-kernels REQUIRED HINTS ${AWKWARD_LIBRARIES})
 find_library(LIBAWKWARD awkward REQUIRED HINTS ${AWKWARD_LIBRARIES})
 find_library(LIBDL dl REQUIRED)
-message(STATUS "Libraries: ${CPU-KERNELS} ${LIBAWKWARD} ${LIBDL}")
+find_library(LIBARROW libarrow.so.400 REQUIRED HINTS ${PYARROW_LIBRARY_DIRS} NO_DEFAULT_PATH)
+find_library(LIBARROW_PYTHON libarrow_python.so.400 REQUIRED HINTS ${PYARROW_LIBRARY_DIRS})
+message(STATUS "Libraries: ${CPU-KERNELS} ${LIBAWKWARD} ${LIBDL} ${LIBARROW} ${LIBARROW_PYTHON}")
+
+
+add_library(pyarrow IMPORTED INTERFACE)
+target_link_libraries(pyarrow INTERFACE ${LIBARROW} ${LIBARROW_PYTHON}) #  ${PYTHON_LIBRARIES})
+target_include_directories(pyarrow INTERFACE "${PYARROW_INCLUDE}")
+target_link_options(pyarrow INTERFACE "-Wl,--unresolved-symbols=ignore-all")
```

Comment:
Not sure why I'm having to manually add this, it's not macOS...

Reply:
Hm. Maybe @kratsg and I can poke more tomorrow on this. If you're having to manually add things that you feel you shouldn't that's probably not great given something we've done. :/

Reply:
I most definitely don't understand this.

Reply:
If you need a symbol, like `PyObject_Repr` (just picking from the long list), then you want to just ignore that the symbol is undefined - when you load this _from_ a Python interpreter, the symbol will already be there (after all, it _is_ CPython, so all Python's C symbols have to be loaded). Normally, on macOS, you have to pass a flag to keep from erroring, but on linux GCC, you don't, that's the default. At least that's what I remembered (and pybind11's code should handle adding the proper flags for you).

I think there's a way to specify "symbols are in this file, but don't use them", but didn't' look that up. I feel like there were issues using it in the past with pybind11, or maybe we were just lazy. It makes the build a little safer by still catching missed links for other symbols besides Python ones.